File: leetcode/Python/maximum-points-you-can-obtain-from-cards.py
class Solution:
    def maxScore(self, cardPoints: List[int], k: int) -> int:
        pSum=sum(cardPoints[-k:])
        sol=pSum
        for i in range(k):
            pSum=pSum+cardPoints[i]-cardPoints[i-k]
            sol=max(sol,pSum)
        return sol

File: leetcode/C++/maximum-points-you-can-obtain-from-cards.cpp
class Solution {
public:
    int maxScore(vector<int>& cardPoints, int k) {
        int sol=0,sum=0, sz=cardPoints.size();
        for(int i=0; i<k; ++i) sum+=cardPoints[sz-i-1];
        sol=sum;
        for(int i=0; i<k; ++i){
            sum+=cardPoints[i]-cardPoints[sz-k+i];
            sol=max(sol,sum);
        }
        return sol;
    }
};